src/Plugin/Commerce/PriceRuleCalculation/AttributePrice.php

```diff
@@ -0,0 +1,227 @@
+ *   entity_type = "commerce_product_variation",
+ * )
+ */
+class AttributePrice extends PriceRuleCalculationBase {
+
+  /**
+   * The attribute price manager.
+   *
+   * @var \Drupal\commerce_price_rule\AttributePriceCalculationManager
+   */
+  protected $attributePriceManager;
+
+  /**
+   * Constructs a new PriceList object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The pluginId for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\commerce_price\RounderInterface $rounder
+   *   The rounder.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    RounderInterface $rounder,
+    EntityTypeManagerInterface $entity_type_manager
+  ) {
+    parent::__construct(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $rounder
+    );
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(
+    ContainerInterface $container,
+    array $configuration,
+    $plugin_id,
+    $plugin_definition
+  ) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('commerce_price.rounder'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'amount' => NULL,
+    ] + parent::defaultConfiguration();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form += parent::buildConfigurationForm($form, $form_state);
+
+    $attributes = $this->getAttributesPriceField();
+    $form['price'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Attribute price field mapping'),
+      '#open' => TRUE,
+    ];
+
+    // Loop through all attributes and display the price fields.
+    foreach ($attributes as $key => $price_fields) {
+      $field = NULL;
+      if (!empty($this->configuration['attribute_price']['price'][$key]['field'])) {
+        $field = $this->configuration['attribute_price']['price'][$key]['field'];
+      }
+      $form['price'][$key] = [
+        '#type' => 'details',
+        '#title' => $key,
+        '#collapsible' => TRUE,
+        '#open' => FALSE,
+      ];
+      $form['price'][$key]['field'] = [
+        '#type' => 'select',
+        '#title' => $this->t('Price Field'),
+        '#options' => $price_fields['commerce_price_fields'],
+        '#description' => $this->t('Select the field from which the attribute price needs to be fetched'),
+        '#default_value' => $field,
+      ];
+    }
```

Comment:
**CHANGE REQUEST**

Since it's only one form element per attribute let's put them all in one `details` container.

Reply:
Done